File: sample/src/main/java/rebus/navutils/sample/ResultActivity.java
package rebus.navutils.sample;

import android.os.Bundle;

import rebus.navutils.NavUtils;
import rebus.navutils.activity.BaseActivity;
import rebus.navutils.fragment.v4.FragmentUtils;

public class ResultActivity extends BaseActivity {

    @Override
    public void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setTitle("NavUtils Utils");
        setNavigationIcon(R.drawable.ic_back);
        setFinishOnNavigationOnClickListener();
        Bundle bundle = new Bundle();
        bundle.putInt(NavUtils.NAV_ANIM, getAnimationType());
        switch (getAnimationType()) {
            case NavUtils.HORIZONTAL_RIGHT:
                FragmentUtils.Builder(getSupportFragmentManager(), R.id.container)
                        .animationType(NavUtils.HORIZONTAL_RIGHT)
                        .noEnterAnimations(true)
                        .arguments(bundle)
                        .replace(getActivity(), ResultFragment.class);
                setSubtitle("ResultActivity NavUtils.HORIZONTAL_RIGHT");
                break;
            case NavUtils.HORIZONTAL_LEFT:
                FragmentUtils.Builder(getSupportFragmentManager(), R.id.container)
                        .animationType(NavUtils.HORIZONTAL_LEFT)
                        .noEnterAnimations(true)
                        .arguments(bundle)
                        .replace(getActivity(), ResultFragment.class);
                setSubtitle("ResultActivity NavUtils.HORIZONTAL_LEFT");
                break;
            case NavUtils.VERTICAL_BOTTOM:
                FragmentUtils.Builder(getSupportFragmentManager(), R.id.container)
                        .animationType(NavUtils.VERTICAL_BOTTOM)
                        .noEnterAnimations(true)
                        .arguments(bundle)
                        .replace(getActivity(), ResultFragment.class);
                setSubtitle("ResultActivity NavUtils.VERTICAL_BOTTOM");
                break;
            case NavUtils.VERTICAL_TOP:
                FragmentUtils.Builder(getSupportFragmentManager(), R.id.container)
                        .animationType(NavUtils.VERTICAL_TOP)
                        .noEnterAnimations(true)
                        .arguments(bundle)
                        .replace(getActivity(), ResultFragment.class);
                setSubtitle("ResultActivity NavUtils.VERTICAL_TOP");
                break;
            case NavUtils.NONE:
                FragmentUtils.Builder(getSupportFragmentManager(), R.id.container)
                        .animationType(NavUtils.NONE)
                        .noEnterAnimations(true)
                        .arguments(bundle)
                        .replace(getActivity(), ResultFragment.class);
                setSubtitle("ResultActivity NavUtils.NONE");
                break;
            case NavUtils.SYSTEM:
                FragmentUtils.Builder(getSupportFragmentManager(), R.id.container)
                        .animationType(NavUtils.SYSTEM)
                        .noEnterAnimations(true)
                        .arguments(bundle)
                        .replace(getActivity(), ResultFragment.class);
                setSubtitle("ResultActivity NavUtils.SYSTEM");
                break;
        }
    }

    @Override
    protected int setLayoutResource() {
        return R.layout.activity_result;
    }

}

File: sample/src/main/java/com/raphaelbussa/navutils/sample/ResultActivity.kt
package com.raphaelbussa.navutils.sample

import android.os.Bundle
import com.raphaelbussa.navutils.NavUtils
import com.raphaelbussa.navutils.activity.BaseActivity
import com.raphaelbussa.navutils.pushFragment

class ResultActivity : BaseActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setTitle("NavUtils Utils")
        setNavigationIcon(R.drawable.ic_back)
        setFinishOnNavigationOnClickListener()
        val bundle = Bundle()
        bundle.putInt("PARENT_ANIM", animationType.anim)
        when (animationType) {
            NavUtils.Anim.HORIZONTAL_RIGHT -> {
                pushFragment(ResultFragment::class) {
                    animationType(animationType)
                    noEnterAnimations(true)
                    arguments(bundle)
                }.replace(R.id.container)
                setSubtitle("ResultActivity NavUtils.Anim.HORIZONTAL_RIGHT")
            }
            NavUtils.Anim.HORIZONTAL_LEFT -> {
                pushFragment(ResultFragment::class) {
                    animationType(animationType)
                    noEnterAnimations(true)
                    arguments(bundle)
                }.replace(R.id.container)
                setSubtitle("ResultActivity NavUtils.Anim.HORIZONTAL_LEFT")
            }
            NavUtils.Anim.VERTICAL_BOTTOM -> {
                pushFragment(ResultFragment::class) {
                    animationType(animationType)
                    noEnterAnimations(true)
                    arguments(bundle)
                }.replace(R.id.container)
                setSubtitle("ResultActivity NavUtils.Anim.VERTICAL_BOTTOM")
            }
            NavUtils.Anim.VERTICAL_TOP -> {
                pushFragment(ResultFragment::class) {
                    animationType(animationType)
                    noEnterAnimations(true)
                    arguments(bundle)
                }.replace(R.id.container)
                setSubtitle("ResultActivity NavUtils.Anim.VERTICAL_TOP")
            }
            NavUtils.Anim.NONE -> {
                pushFragment(ResultFragment::class) {
                    animationType(animationType)
                    noEnterAnimations(true)
                    arguments(bundle)
                }.replace(R.id.container)
                setSubtitle("ResultActivity NavUtils.Anim.NONE")
            }
            NavUtils.Anim.SYSTEM -> {
                pushFragment(ResultFragment::class) {
                    animationType(animationType)
                    noEnterAnimations(true)
                    arguments(bundle)
                }.replace(R.id.container)
                setSubtitle("ResultActivity NavUtils.Anim.SYSTEM")
            }
        }
    }

    override fun setLayoutResource() = R.layout.activity_result

}
